applications/CompressiblePotentialFlowApplication/custom_elements/transonic_perturbation_potential_flow_element.cpp

```diff
@@ -1039,6 +1164,46 @@ void TransonicPerturbationPotentialFlowElement<TDim, TNumNodes>::SelectUpwindEle
     }
 }
 
+template<int TDim, int TNumNodes>
+int TransonicPerturbationPotentialFlowElement<TDim, TNumNodes>::GetAdditionalNode() const
```

Comment:
Really like this function! I have some suggestions related to the fact that I think you can reuse GetSortedIds that you implemented in your last PR and the find function from std to gain in efficiency and reduce the number of lines.